core/matcher.go

```diff
@@ -63,25 +63,62 @@ RuleSearch:
 }
 
 // getProccessedInputAndHitValue gets the processed input from the message input and the true/false if it was a successfully hit rule.
-func getProccessedInputAndHitValue(messageInput, ruleRespondValue, ruleHearValue string) (string, bool) {
+func getProccessedInputAndHitValue(message models.Message, rule models.Rule) (string, bool) {
 	processedInput, hit := "", false
-	if ruleRespondValue != "" {
+
+	ruleRespondValue := rule.Respond
+	ruleHearValue := rule.Hear
+
+	if rule.Respond != "" {
+		log.Debug().Msgf("Respond Rule %s", rule.Name)
+
+		messageInput := message.Input
 		processedInput, hit = utils.Match(ruleRespondValue, messageInput, true)
-	} else if ruleHearValue != "" { // Are we listening to everything?
+	} else if rule.Hear != "" { // Are we listening to everything?
+		log.Debug().Msgf("HearRule %s", rule.Name)
+
+		messageInput := message.Input
 		_, hit = utils.Match(ruleHearValue, messageInput, false)
+	} else if rule.ReactionsAdded != "" {
+		log.Debug().Msgf("ReactionAdded Rule %s", rule.Name)
+
+		messageReaction := message.ReactionAdded
+		processedInput, hit = utils.Match(rule.ReactionsAdded, messageReaction, false)
+	} else if rule.ReactionsRemoved != "" {
+		log.Debug().Msgf("ReactionRemoved Rule %s", rule.Name)
+
+		messageReaction := message.ReactionRemoved
+		processedInput, hit = utils.Match(rule.ReactionsRemoved, messageReaction, false)
```

Comment:
let's remove the debug log messages in this section. it basically prints an inventory of all rules every time a message gets processed. probably not a bad idea to do on startup where we already print the number of rules that you have setup (but not necessary to add in this PR).

Reply:
Removed